### DISCUSSION TITLE
Example 12 setVal() not automatically updating Value in Home App

### DISCUSSION THREAD

**Discussion by VirtualSquash** [HomeSpan/HomeSpan#764] [Q&A] [answered]
Hi there,
I am currently working through the examples and have a question about example 12. The values from the temperature and air quality sensors are not automatically updated in my Home app (Sometimes it does it ones after opening the app). I have to close and re-open the app to update the values. I completely reset the ESP32 and Homespan but it did not change. It seems that my iPhone gets disconnected after about a second. Do I need a HomePod or AppleTV for this to work?

I am using:
ESP32 DevKit V1
iPhone 15 Pro with iOS 17.2.1
iPhone, ESP32 and router are all next to each other.

![image](https://github.com/HomeSpan/HomeSpan/assets/48453234/70607fe3-1373-44a3-815c-7f8b39612c2b)

![image](https://github.com/HomeSpan/HomeSpan/assets/48453234/a399148c-8332-4d06-8e91-32ec1476622d)

**Comment by HomeSpan** [comment, marked as answer]
Starting at about iOS 16 Apple now requires a HomePod or AppleTV to be used as a Home Hub to receive notifications, such as those from a temperature sensor.  A direct iPhone connection will only last a short while before disconnecting.

**Reply by VirtualSquash** [reply]
Thanks for the answer and also Thanks for the Library.
I got a spare AppleTV so this should not be a problem.

**Reply by HomeSpan** [reply]
Note AppleTv must be generation 4 or later.

**Reply by VirtualSquash** [reply]
👍, I bought a used Gen4 AppleTV a while ago for another project because it can be powered from 12V.
